src/algebra/algebra/basic.lean

```diff
@@ -526,10 +526,6 @@ lemma map_finsupp_sum {α : Type*} [has_zero α] {ι : Type*} (f : ι →₀ α)
 @[simp] lemma map_bit1 (x) : φ (bit1 x) = bit1 (φ x) :=
 φ.to_ring_hom.map_bit1 x
 
```

Comment:
I agree the other `subsingleton` lemma is more general, but this one seems to be a lot more useful in practice (as we saw in #8110). So can we keep this lemma for ease of use (at least until we fix #6025)?

```suggestion

/-- Convenience lemma for a specific case of `alg_hom.subsingleton`. -/
-- TODO[gh-6025]: once `alg_hom.subsingleton` and `subalgebra.subsingleton_of_subsingleton`
-- can be instances, this could be removed
lemma subsingleton_left [subsingleton A] : subsingleton (A →ₐ[R] B) :=
⟨λ f g, alg_hom.ext $ λ x, subsingleton.elim 0 x ▸ f.map_zero.trans g.map_zero.symm⟩
```

Reply:
I'm not sure that makes sense to add. How about the comment I suggest in https://github.com/leanprover-community/mathlib/pull/8140/files#r661322470?